e2e/arm.test.ts

```diff
@@ -0,0 +1,186 @@
+		serviceHost: testConfig.signalingAddress,
+		credentials: {
+			type: 'api-key',
+			authEntity: API_KEY_ID,
+			payload: API_KEY,
+		},
+	}
+
+	const client = await createViamClient(opts)
+	return client
+}
+
+let viamClient: ViamClient
+
+test.beforeAll(async () => {
+	viamClient = await connectViamClient()
+})
+
+const armConfig = {
+	components: [
+		{
+			name: 'arm-1',
+			api: 'rdk:component:arm',
+			model: 'rdk:builtin:fake',
+			attributes: {
+				'arm-model': 'ur5e',
+			},
+			frame: {
+				parent: 'world',
+				translation: {
+					x: 0,
+					y: 0,
+					z: 0,
+				},
+				orientation: {
+					type: 'ov_degrees',
+					value: {
+						x: 0,
+						y: 0,
+						z: 1,
+						th: 0,
+					},
+				},
+			},
+		},
+		{
+			name: 'generic-1',
+			api: 'rdk:component:generic',
+			model: 'rdk:builtin:fake',
+			attributes: {},
+			frame: {
+				parent: 'arm-1',
+				translation: {
+					x: 0,
+					y: 0,
+					z: 100,
+				},
+				orientation: {
+					type: 'ov_degrees',
+					value: {
+						x: 0,
+						y: 0,
+						z: 1,
+						th: 0,
+					},
+				},
+				geometry: {
+					type: 'sphere',
+					r: 100,
+				},
+			},
+		},
+	],
+}
+
+test('arm', async ({ browser }) => {
+	const testPrefix = 'ARM'
+	await viamClient.appClient.updateRobotPart(
+		testConfig.partId,
+		testConfig.name,
+		Struct.fromJson(armConfig as unknown as JsonValue)
+	)
+
+	const failedScreenshots = [] as string[]
+	const context = await browser.newContext()
+	await context.addCookies([
+		{
+			name: 'weblab_experiments',
+			value: 'MOTION_TOOLS_RENDER_ARM_MODELS',
+			domain: 'localhost',
+			path: '/',
+		},
+	])
+	const page = await context.newPage()
+	await page.waitForTimeout(5000)
+	page.on('console', (message) => {
+		console.log(`[${message.type()}] ${message.text()}`)
+	})
+	await page.goto('/')
+	await expect(page.getByText('World', { exact: true })).toBeVisible()
+
+	// SETUP CONFIG
+	await expect(page.getByTestId('icon-robot-outline')).toBeVisible()
+	await page.getByTestId('icon-robot-outline').click()
+
+	await expect(page.getByText('Add config', { exact: true })).toBeVisible()
+	await page.getByText('Add config', { exact: true }).click()
+
+	await expect(page.getByPlaceholder('Host')).toBeVisible()
+	await page.getByPlaceholder('Host').fill(testConfig.host)
+	await expect(page.getByPlaceholder('Part ID')).toBeVisible()
+	await page.getByPlaceholder('Part ID').fill(testConfig.partId)
+	await expect(page.getByPlaceholder('API Key ID')).toBeVisible()
+	await page.getByPlaceholder('API Key ID').fill(testConfig.apiKeyId)
+	await expect(page.getByPlaceholder('API Key Value')).toBeVisible()
+	await page.getByPlaceholder('API Key Value').fill(testConfig.apiKeyValue)
+	await expect(page.getByPlaceholder('Signaling Address')).toBeVisible()
+	await page.getByPlaceholder('Signaling Address').fill(testConfig.signalingAddress)
+
+	await page.getByTestId('icon-close').click()
+
+	await page.waitForTimeout(5000) // wait for arm geometries to load
```

Comment:
Can we instead use `waitForSelector` and wait for the geometries to appear in the treeview?